l1-contracts/contracts/governance/L2AdminFactory.sol

```diff
@@ -39,4 +43,15 @@ contract L2AdminFactory {
 
         admin = address(new ChainAdmin{salt: _salt}(restrictions));
     }
+
+    function _validateZeroAddress(address[] memory _restrictions) internal view {
```

Comment:
I don't think we should add this check.  `ChainAdmin` constructor already checks on zero address

Reply:
maybe let's still check it here for failing early + it does not cost that much

Reply:
It i a nit either way, but I think it is better to remove redundant code, also you would need to document `_validateZeroAddress` function

Reply:
I'll just document it, similar to M2